discord/poll.py

```diff
@@ -0,0 +1,644 @@
+
+            elif isinstance(self.emoji, PartialEmoji):
+                payload['emoji'] = self.emoji.to_dict()  # type: ignore
+
+            else:
+                payload['emoji'] = {'name': str(self.emoji)}
+
+        return payload
+
+    @classmethod
+    def from_dict(cls, *, data: PollMediaPayload) -> PollMedia:
+        """Returns a new instance of this class from a payload."""
+
+        return cls(text=data['text'], emoji=PartialEmoji.from_dict(data.get('emoji', {})))
+
+    def __iter__(self) -> Union[str, Optional[PollMediaEmoji]]:
+        yield self.text
+        yield self.emoji
+
+    def __repr__(self) -> str:
+        return f'<PollMedia text={self.text} emoji={self.emoji}>'
+
+
+class PollAnswerBase:
+
+    if TYPE_CHECKING:
+        id: int
+        _message: Optional[Message]
+        _state: Optional[ConnectionState]
+
+    async def users(self, *, after: Snowflake = MISSING, limit: int = 25) -> AsyncIterator[Union[User, Member]]:
+        r"""|coro|
+
+        Retrieves all the voters of this answer.
+
+        .. warning::
+
+            This can only be called when the poll is accessed via :attr:`Message.poll`.
+
+        Parameters
+        ----------
+        after: :class:`Snowflake`
+            Fetches users after this ID.
+        limit: :class:`int`
+            The max number of users to return. Can be up to 100.
+
+        Raises
+        ------
+        HTTPException
+            Retrieving the users failed.
+
+        Yields
+        ------
+        Union[:class:`User`, :class:`Member`]
+            The users that voted for this poll answer. This can be a :class:`Member` object if the poll
+            is in a guild-message context, for other contexts it will always return a :class:`User`, or when
+            the member left the guild.
+        """
+
+        # As this is the same implementation for both PollAnswer objects
+        # we should just recycle this.
+
+        if not self._state:
+            raise RuntimeError('You cannot fetch users in a non-message-attached poll')
+
+        data = await self._state.http.get_poll_answer_voters(
+            self._message.channel.id,
+            self._message.id,
+            self.id,
+            after.id if after is not MISSING else MISSING,
+            limit
+        )
+
+        if not self._message.guild:
+            for user in data.get('users'):
+                yield User(state=self._state, data=user)
+
+        else:
+            guild = self._message.guild
+
+            for user in data.get('users'):
+                member = guild.get_member(int(user['id']))
+
+                yield member or User(state=self._state, data=user)
+
+
+class PollAnswerCount(PollAnswerBase):
+    """Represents a poll answer count.
+
+    This is not meant to be user-constructed but instead obtained by the results in
+    :attr:`Poll.answer_counts`
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The answer ID.
+    me_voted: :class:`bool`
```

Comment:
I know this parameter name comes from the API payload but I personally find it the naming a bit lacking. How about `self_vote` or something along those lines?

Reply:
Yeah, will consider it 🤔